chapter.md

```diff
@@ -0,0 +1,781 @@
+    replacing the specified property with the specified value as it does so.
+    (Think of this as "make me a copy of X, but with Y set to Z".)
+*   `merge`: create a copy of one object with properties merged in from a second object.
+    (This is like `assocPath`, but using a second object to get multiple changes at once.)
+*   `path`: retrieve the value at a specified location in a structured object.
+
+
+With these helpers in hand,
+our `formReducer` looks like this:
+
+FIXME: @renvrant can you please check against https://github.com/renvrant/formcontrol-freaks/pull/3#discussion_r106822741
+
+```ts
+import { path, assocPath, merge } from 'ramda';
+
+export function formReducer(state = initialState: IForm, action) {
+  switch (action.type) {
+
+  case 'SAVE_FORM':
+     return assocPath(
+       action.payload.path,
+       merge(path(action.payload.path, state), action.payload.value),
+       state
+     );
+
+  default:
+    return state;
+  }
+}
+```
+
+There is actually less going on here than first appears.
+Working from the outside in,
+if the action's type is *not* `SAVE_FORM`,
+then `formReducer` returns the original state without any changes.
+This means that we can safely chain this reducer together
+with others that handle other parts of our user interface:
+as long as they all use distinct keys for their actions,
+they will watch the state fly by without doing anything we don't want them to.
+
+If the action does have the right type,
+`formReducer` uses `path` to get the part of the state that needs to be updated,
+then uses `merge` and `assocPath` to create a shallow copy of the state,
+replacing that element,
+*and only that element*,
+with a different value.
+The most important word in the previous sentence is "create":
+at this point,
+the state that was passed in is thrown away and a new one created.
+Some parts of the old state are recycled–`assocPath` does a shallow clone–but
+those are the parts that *weren't* changed,
+and if other reducers do their jobs properly,
+they will replace those parts rather than updating them in place
+when it's their turn to act.
+
+## Connecting the Dots
+
+It's now time to turn our attention to the interface
+that will trigger this state change
+and reflect any changes to the state triggered by other interface components.
+Since we're using Angular 2,
+we will define a class
+and use the `@Component` decorator to weld some metadata to it:
+
+```ts
+@Component({
+  selector: 'character-form',
+  template: require('./character-form.html')
+})
+export class CharacterForm {
+  @ViewChild(NgForm) ngForm: NgForm;
+  public characterForm;
+  private formSubs;
+
+  constructor(private ngRedux: NgRedux<IAppState>) {}
+}
+```
+
+There's a lot going on here,
+so we'll start with a high-level explanation and then dive into details.
+As the diagram shows,
+Angular will keep `ngForm` and `characterForm` in sync with each other
+using its own dark magic.
```

Comment:
Perhaps mention we will get to said dark magic later rather than leaving it here?